### ISSUE TITLE
LICENSE missing

### ISSUE THREAD

**Issue by oheim** [ENSTABretagneRobotics/VIBES#93]
The LICENSE file is empty and the source code contains no copyright statements. What are the conditions for use, copy, modification and redistribution of VIBES?

**Comment by nicolaje** [comment]
Dear @oheim , indeed we haven't really discussed licensing yet. It is definitely something we have to address.
In the meantime, could you tell us about your application and why type of licensing would suit you best?
Thank you

**Comment by oheim** [comment]
For now I just need permission to legally use VIBES during the [IAMOOC](http://iamooc.ensta-bretagne.fr/). I would prefer to have a [free (GPL compatible) license](http://www.gnu.org/licenses/license-list#GPLCompatibleLicenses).

Another possibly future use case would be to interface VIBES with the [Interval package for Octave](http://octave.sourceforge.net/interval/) for plotting. This would also require a free license.

**Comment by oheim** [comment]
I can provide a client-api for Octave as a loadable Octave package, which uses the native plugin interface (oct-files) and is build upon the c++ client api. This client-api must be released under GPL compatible terms, see http://wiki.octave.org/FAQ#Licensing_issues

Should the Octave client-api be compatible with the MATLAB client-api?

**Comment by dvinc** [comment]
I think we should release the client-api files as GPL. I will start preparing C++ API with a GPLv3 license, as this is needed to merge #98. I will need every API contributor (@nicolaje, @benEnsta, @SimonRohou and myself) to state he agrees to release the client-api files he contributed to in GPLv3.

I think the licensing for the viewer should be discussed separately (e.g. do we add conditions for commercial, academical use, etc...). I opened #99 to discuss this topic.

**Comment by nicolaje** [comment]
I think GPL might limit adoption of VIBes in the industry.
The MIT/X License is more business friendly, so my preference would go toward MIT License.
To my understanding, it won't prevent @oheim to license the Octave API under GPL3, since MIT is GPL compatible.

**Comment by oheim** [comment]
@nicolaje, correct. You should not put the C++ client-api under GPL, if you don't want to restrict “industry users (*)” bundling it. The C++ client API is a good candidate for bundling. A GPL compatible license would be perfectly okay for use in the Octave API.

You could also dual license the C++ client-api under GPL and something else.

(*) The term “industry users” means software vendors who want to bundle the C++ client API in their product, which the distribute under proprietary license terms.

**Comment by dvinc** [comment]
I agree with your point of view @nicolaje, you are right the GPLv3 I proposed for the API is by far too restrictive for business usage, since every derived work would have to be GPLed. We should select a license compatible with most licenses.
I think we'll have to provide a license for each API, since, e.g., the Octave version, as a derived work of the GPLed octave native plugin interface, will be GPL.

**Comment by oheim** [comment]
Could you resolve the state of the program's license already?

**Comment by black-puppydog** [comment]
Friendly bump :)
I too used VIBES for the IAMOOC, and just realized now it doesn't actually have a license, which means that it is currently only usable in a very restricted sense governed by the github ToS. I think it's fair to say that's much more harmful to industry users than any other license. :P

It would be really nice to see this being agreed on while the set of contributors is still small (seven currently).

In any case: thanks for the work on this! :)

**Comment by SimonRohou** [comment]
VIBes is under LGPLv3 license since 73cea9ad0cee6885be175e3e5ee9322bd4812546.
